src/Interpreters/Context.cpp

```diff
@@ -807,7 +851,7 @@ Strings Context::getWarnings() const
 /// TODO: remove, use `getTempDataOnDisk`
 VolumePtr Context::getGlobalTemporaryVolume() const
 {
-    auto lock = getLock();
+    auto lock = getGlobalLock();
```

Comment:
@kitaisreal Hi, why here is `getGlobalLock()` instead of `getGlobalSharedLock()`?

Reply:
When I added this pull request I did not have statics checks in place, and it was safer to add exclusive lock.
After https://github.com/ClickHouse/ClickHouse/pull/55278, I replaced most of the exclusive locks to shared locks, because it is much safer now. If I missed something feel free to create pull request.
For those specific place https://github.com/ClickHouse/ClickHouse/pull/60997.